libs/screen/image.cpp

```diff
@@ -1068,6 +1068,15 @@ void _blitRow(Image_ img, int xy, Image_ from, int xh) {
     blitRow(img, XX(xy), YY(xy), from, XX(xh), YY(xh));
 }
 
+void blit(Image_ dst, int xDst, int yDst, int wDst, int hDst, Image_ src, int xSrc, int ySrc, int wSrc, int hSrc, bool transparent) {
+    // TODO
+}
+
+//%
+void _blit(Image_ dst, int xyDst, int whDst, Image_ src, int xySrc, int whSrc, bool transparent) {
```

Comment:
there is 4 argument limit on C++ functions (which is why there's this XX/YY macros)

I think the fastest way to pass the arguments would be an array (you can allocate just one array); probably just stick the numbers in the array and pass the images separately

Reply:
I was wondering why these parameters were packed!

Reply:
@mmoskal How do I test the cpp side? Does it need to round-trip to the compiler backend? Use uploadtrg?

Reply:
ARM passes first 4 args in registers; I never implemented passing args on stack...

Reply:
@eanders-MS you can just test it using local serve! Just don't use --rb

Reply:
I typically build from command line with `pxt` in `pxt-arcade/projects/myproject`; you need something like the following in `pxt.json`: 
```json
    "dependencies": {
        "hw---stm32f401": "file:../../libs/hw---stm32f401",
        "device": "file:../../libs/device"
    },
```

I have the following env variables set:
```json
declare -x PXT_ASMDEBUG="1"
declare -x PXT_COMPILE_SWITCHES="size"
declare -x PXT_FORCE_LOCAL="yes"
declare -x PXT_NODOCKER="yes"
declare -x PXT_RUNTIME_DEV="yes"
```

Check out `pxt help` for descriptions.

Reply:
Where would I declare the static array? I don't quite see how it would be shared between TS and c++ implementations.

Reply:
Declare in TypeScript and pass as argument of type RefCollection to c++